src/Mod/Path/PathScripts/PathDeburr.py

```diff
@@ -51,7 +51,14 @@ def translate(context, text, disambig=None):
 
 
 def toolDepthAndOffset(width, extraDepth, tool):
-    '''toolDepthAndOffset(width, extraDepth, tool) ... return tuple for given parameters.'''
+    '''toolDepthAndOffset(width, extraDepth, tool) ... return tuple for given\n
+       parameters.'''
+
+    if not (hasattr(tool, 'CuttingEdgeAngle')
+            and hasattr(tool, 'FlatRadius')
+            and hasattr(tool, 'Diameter')):
+        raise ValueError('Deburr requires tool with flatradius, diameter, and CuttingEdgeAngle\n')
+
```

Comment:
This prevents the use of endmills which was supported somewhat intentionally. Not sure if it's still needed now that there is profiling along edges - haven't used that though.

Reply:
This fix was just to catch the error that was painting the console red.  Has an earlier patch eliminated the user of endmills?

Reply:
Reworked the attribute checking and default to allow toolbit endmills as well as legacy endmills